consensus/misc/eip2935.go

```diff
@@ -0,0 +1,42 @@
+package misc
+
+import (
+	// "fmt"
+	"math/big"
+
+	"github.com/holiman/uint256"
+
+	"github.com/ledgerwatch/erigon-lib/chain"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/consensus"
+	"github.com/ledgerwatch/erigon/core/state"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/params"
+)
+
+func StoreBlockHashesEip2935(header *types.Header, state *state.IntraBlockState, config *chain.Config, headerReader consensus.ChainHeaderReader) {
+	if header.Number.Cmp(big.NewInt(0)) == 0 { // Activation of fork at Genesis
+		return
+	}
+	parent := headerReader.GetHeaderByHash(header.ParentHash)
+	_storeHash(parent.Number, header.ParentHash, state)
+	// If this is the fork block, add the parent's direct `HISTORY_SERVE_WINDOW - 1` ancestors as well
+	if parent.Time < config.PragueTime.Uint64() {
+		p := parent.Number.Uint64()
+		window := params.BlockHashHistoryServeWindow - 1
+		if p < window {
```

Comment:
It's probably equivalent, but I find the pseudo-code in the spec easier to read. What I mean that instead of this `p < window` branch you have a early break in the loop. Something like:
```
ancestorNumber := parent.Number.Uint64() - 1
if ancestorNumber == 0 { // stop at genesis block
  break
}
storeHash(big.NewInt(ancestorNumber), parent.ParentHash, state)
...
}

Reply:
It's an optimization to not have to do `big.Int` to `uint64()`, or any `big.Int` operation  that many times

Reply:
I see. Between optimization and clarity I'd choose clarity until profiling confirms that the performance degradation is material, but OK.